### ISSUE TITLE
Line number of error is incorrect

### ISSUE THREAD

**Issue by dariober** [snakemake/snakemake#1256]
**Snakemake version**: 6.10.0 installed via bioconda/mamba

**Describe the bug**

When a snakefile contains a syntax error or some other bug,  snakemake reports the wrong line number of the error. More specifically, it seems to me that:

* The reported line number is consistently after the actual line number
* As the Snakefile becomes longer, the difference between reported and actual line number grows
* Earlier versions do not show this bug (e.g. v5.4.3 works)

This is quite a problem since it makes it difficult to locate bugs!

**Minimal example**

Here the syntax error is on **line 9**:

```
rule all:
    input:
        'done.txt',
        
rule trim_transcripts:
    output:
        done.txt,
    run:
        if 1 = 2:  # Syntax error: It should be 1 == 2
            print('foo')
```

Snakemake 6.10.0 reports the error on **line 18**:

```
snakemake -j 1 -n -s Snakefile2
SyntaxError in line 18 of /export/III-data/wcmp_bioinformatics/db291g/git_repos/wcip/20211109_kasia_asaj_snp/Snakefile2:
invalid syntax (Snakefile2, line 18)
```

Snakemake 5.4.3 is correct:

```
/export/projects/III-data/wcmp_bioinformatics/db291g/miniconda3/envs/20190312_andy_alba3_motifs/bin/snakemake -j 1 -n -s Snakefile2
SyntaxError in line 9 of /export/III-data/wcmp_bioinformatics/db291g/git_repos/wcip/20211109_kasia_asaj_snp/Snakefile2:
invalid syntax
```

**Comment by Linkrw12** [comment]
This also exists in Snakemake 6.12.1

**Comment by dariober** [comment]
I think it's been fixed in v6.12.3 (thanks!) - please correct me if I'm wrong...

**Comment by yardimcilab** [comment]
This error persists in version 8.0.1.
